iothub/device/samples/how to guides/DeviceReconnectionSample/ExceptionHelper.cs

```diff
@@ -29,34 +29,71 @@ internal class ExceptionHelper
             typeof(WebException),
             typeof(WebSocketException),
         };
-
-        private static bool IsNetwork(Exception singleException)
-        {
-            return s_networkExceptions.Any(baseExceptionType => baseExceptionType.IsInstanceOfType(singleException));
-        }
+        private static readonly bool s_isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
 
         internal static bool IsNetworkExceptionChain(Exception exceptionChain)
         {
-            return exceptionChain.Unwind(true).Any(e => IsNetwork(e) && !IsTlsSecurity(e));
+            return Unwind(exceptionChain, true).Any(e => IsNetwork(e) && !IsTlsSecurity(e));
         }
 
-        internal static bool IsSecurityExceptionChain(Exception exceptionChain)
+        private static bool IsNetwork(Exception singleException)
         {
-            return exceptionChain.Unwind(true).Any(e => IsTlsSecurity(e));
+            return s_networkExceptions.Any(baseExceptionType => baseExceptionType.IsInstanceOfType(singleException));
         }
 
         private static bool IsTlsSecurity(Exception singleException)
```

Comment:
I don't love having a class like this in our sample code. Maybe we should consider making flags in our IotHubClientException class that tells the user this information.

Reply:
Not in this PR though, of course

Reply:
I agree, but I think any improvements on this front should be saved for v2.

@brycewang-microsoft do we have any new logic in v2 to determine if transient based on what we have here?

Reply:
No, I don't think so. The `IsTransient` flag in "IotHubClientException" is determined based on 
1) the `ErrorCode` when it is set 
OR 
2) how `IsTransient` was set in the original "[IotHubException](https://github.com/Azure/azure-iot-sdk-csharp/blob/283631c16e93bdcd4f63b44bd7fcc54e61e8a670/iothub/device/src/Common/Exceptions/IotHubException.cs)" which is the predecessor of "IotHubClientException"

Reply:
@abhipsaMisra when do we think we'd see an exception (or perhaps an inner exception) that we'd want the SDK to check for this condition and use that to set IsTransient to true?

Reply:
The only scenario I'd expect the SDK to run into a security exception is if it was thrown by the TLS stack underneath, i.e. something off about the certificate we're using, or if the certificate store wasn't accessible etc. I'd expect us to run into this during the initial connect, so if we wrap the OpenAsync call with this logic and transform it into an AuthenticationException in there, then we could get rid of this here. (Our SDK currently owns only an UnauthorizedException, but not an AuthenticationException type).